File: app/src/main/java/com/commit451/alakazam/sample/MainActivity.java
package com.commit451.alakazam.sample;

import android.animation.ValueAnimator;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.commit451.alakazam.Alakazam;
import com.commit451.alakazam.EaseInOutBezierInterpolator;
import com.commit451.alakazam.HideRunnable;
import com.commit451.alakazam.ShowRunnable;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final ImageView icon = (ImageView) findViewById(R.id.icon);
        findViewById(R.id.animate).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (icon.getVisibility() == View.VISIBLE) {
                    icon.animate().alpha(0.0f).withEndAction(new HideRunnable(icon));
                } else {
                    icon.animate().alpha(1.0f).withStartAction(new ShowRunnable(icon));
                }
            }
        });

        final Button buttonAnimateMe = (Button) findViewById(R.id.animate_me);
        buttonAnimateMe.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                toggleText(buttonAnimateMe);
                int currentWidth = buttonAnimateMe.getWidth();
                buttonAnimateMe.measure(View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED),
                        View.MeasureSpec.makeMeasureSpec(buttonAnimateMe.getHeight(), View.MeasureSpec.UNSPECIFIED));
                int pendingWidth = buttonAnimateMe.getMeasuredWidth();
                ValueAnimator animator = Alakazam.widthAnimator(buttonAnimateMe, currentWidth, pendingWidth);
                animator.setDuration(800);
                animator.setInterpolator(new EaseInOutBezierInterpolator());
                animator.start();
            }
        });
        toggleText(buttonAnimateMe);
    }

    private void toggleText(TextView textView) {
        if (textView.getText().toString().equals("hi there hows it going today?")) {
            textView.setText("hi");
        } else {
            textView.setText("hi there hows it going today?");
        }
    }
}

File: app/src/main/java/com/commit451/alakazam/sample/MainActivity.kt
package com.commit451.alakazam.sample

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import com.commit451.alakazam.EaseInOutBezierInterpolator
import com.commit451.alakazam.fadeIn
import com.commit451.alakazam.fadeOut
import com.commit451.alakazam.widthAnimator

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val icon = findViewById<ImageView>(R.id.icon)

        findViewById<View>(R.id.animate).setOnClickListener {
            if (icon.visibility == View.VISIBLE) {
                icon.fadeOut()
            } else {
                icon.fadeIn()
            }
        }

        val buttonAnimateMe = findViewById<Button>(R.id.animate_me)
        buttonAnimateMe.setOnClickListener {
            toggleText(buttonAnimateMe)
            val currentWidth = buttonAnimateMe.width
            buttonAnimateMe.measure(View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED),
                    View.MeasureSpec.makeMeasureSpec(buttonAnimateMe.height, View.MeasureSpec.UNSPECIFIED))
            val pendingWidth = buttonAnimateMe.measuredWidth
            val animator = buttonAnimateMe.widthAnimator(currentWidth, pendingWidth)
            animator.duration = 800
            animator.interpolator = EaseInOutBezierInterpolator()
            animator.start()
        }
        toggleText(buttonAnimateMe)
    }

    private fun toggleText(textView: TextView) {
        if (textView.text.toString() == "hi there hows it going today?") {
            textView.text = "hi"
        } else {
            textView.text = "hi there hows it going today?"
        }
    }
}
